### PR TITLE
add ban

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

File: core/controller/sync.ts [hunk 1]
```diff
@@ -82,17 +83,17 @@ export class SyncController extends BaseController {
             }
             const { isExist, peer = null } = responseCommonBlocks.data;
             if (!isExist) {
+                PeerRepository.ban(lastPeerRequested);
```
Comment (Verbalist) [inline core/controller/sync.ts]: lastPeerRequested can be null

Comment (bogdan34) [reply core/controller/sync.ts]: I have already fixed it

File: core/repository/peer.ts [hunk 2]
```diff
@@ -63,6 +66,24 @@ export class PeerRepo {
         return false;
     }
 
+    ban(peer: Peer) {
+        if (!this.has(peer)) {
+            return;
+        }
+        logger.debug(`[Repository][Peer][ban] peer ${peer.ip}:${peer.port} has been banned`);
+        peer = this.getPeerFromPool(peer);
```
Comment (axelsheva) [inline core/repository/peer.ts]: peer = peer?

Comment (bogdan34) [reply core/repository/peer.ts]: yes

**Pull request merged**